File: benchmarks/cases/collatz/collatz.cpp
#include <cstdio>

int main() {
    long n = 1000000;
    long total = 0;

    for (long i = 1; i <= n; i++) {
        long x = i;
        while (x != 1) {
            if (x % 2 == 0) {
                x = x / 2;
            } else {
                x = 3 * x + 1;
            }
            total = total + 1;
        }
    }

    printf("%ld\n", total);
    return 0;
}

File: benchmarks/cases/collatz/collatz.go
package main

import "fmt"

func main() {
	var n int64 = 1000000
	var total int64 = 0

	for i := int64(1); i <= n; i++ {
		x := i
		for x != 1 {
			if x%2 == 0 {
				x = x / 2
			} else {
				x = 3*x + 1
			}
			total++
		}
	}

	fmt.Println(total)
}
